### ISSUE TITLE
Possibility of refactoring/extracting getRepositories?

### ISSUE THREAD

**Issue by ljharb** [ljharb/repo-report#50]
The combination of https://github.com/ljharb/repo-report/blob/main/src/commands/detail.js#L41 and https://github.com/ljharb/repo-report/blob/main/src/utils.js#L111 is what's used to get a list of all the repos I care about.

First, now that we only have one command left (#48), we can probably refactor so that `getRepositories` is one single function.

Second, there are other use cases where I'll want to perform an action on "every repo i care about" - so it might be nice to extract that function into a separate package, so other things can reuse it.

**Comment by Riyabelle25** [comment]
@ljharb is this what you'd been talking about before - i.e adding a flag in `repo-report` to output all the repositories returned by func`getRepositories`?

**Comment by ljharb** [comment]
Yes, exactly.

**Comment by Riyabelle25** [comment]
Right, I've coded in a new option `names` alias `n` that does this. Shall I make a PR mentioning this issue?

**Comment by ljharb** [comment]
I don’t think it needs a short option, just the long one, but yes, that sounds great :-)
